Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+
+**Return value:**
+Array containing splitted values.
+
+**Example:**
+```
+let str = 'It iS a 5r&e@@t Day.'
+let array = str.split(" ");
+console.log(array);
+```
+*Output:*
+
+    [It,iS,a,5r&e@@t,Day.]
+
+### [3. toLowerCase](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase)
+
+**Description:**
+Converts a string to lowercase letters.
+
+**Syntax:**
+ * _string.toLowerCase()_
+
+**Return value:**
+A string, representing the value of a string converted to lowercase.
+
+**Example:**
+```   
+const str = 'It iS a Great Day.';
+const string = str.toLowerCase();
+console.log(string);
+```
+*Output:*
+
+    it is a great day.
+
+### [4. trim](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/trim)
+
+**Description:**
+Removes whitespace from both sides of a string.
+
+**Syntax:**
+ * _string.trim()_
+
+**Return value:**
+String, representing the string with removed whitespace from both ends
+
+**Example:**
+```
+const str = "GeeksforGeeks      ";
+const st = str.trim();
+console.log(st);
+```
+*Output:*
+   
+    GeeksForGeeks
+
+### [5. charAt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charAt)
+
+**Description:**
+Returns the character at the specified index in a string.
+
+**Syntax:**
+ * _string.charAt(index)_
+
+**Return value:**
+String, representing the character at the specified index, or an empty string if the index number is not found
+**Example:**
+```
+const str = 'JavaScript is object oriented language';
+console.log(str.charAt(9));
+```  
+  *Output:*
+
+    t
+    
+## Many forms of for loops
+
+### [1. for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)
+
+**Description:**
+ Loops through the properties of an object. The block of code inside the loop will be executed once for each property.
+
+**Syntax:**
+ * _for (var in object) {<br>
+  code block to be executed<br>
+ }_
+
+| Parameter | Description |
+| :---      | :---        |
+|var	|Required. A variable that iterates over the properties of an object|
+|object	| Required. The specified object that will be iterated|
+
+**Example:**
+ ```  
+ onst array1 = ['a', 'b', 'c'];
+ for (const element of array1) {
+ 	console.log(element);
+ }
+```
+*Output:*
+
+    "a"
+    "b"
+    "c"
+
+### [2.for...of](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...of)
+
+**Description:**
+Loops through the values of an iterable objects.
+
+**Syntax:**
+ * _for (variable of iterable) {<br>
+  // code block to be executed<br>
+}_
+
+| Parameter | Description |
+| :---      | :---        |
+| variable | For every iteration the value of the next property is assigned to the variable. Variable can be declared with const, let, or var.|
+|iterable | An object that has iterable properties.|

```

Comment:
These are language constructs, not functions. This kind of doc isn't necessary. Please remove this

Reply:
Sir, are you referring to the table?